go/consensus/cometbft/api/context.go

```diff
@@ -207,6 +207,14 @@ func (c *Context) CallerAddress() staking.Address {
 	return c.callerAddress
 }
 
+// CallDepth returns the call depth.
+func (c *Context) CallDepth() int {
+	if c.parent == nil {
+		return 0
+	}
+	return c.parent.CallDepth() + 1
```

Comment:
Could store parents call depth in the constructor, to avoid recursing through all parents, but probably doesn't really matter since the depth is limited to a low number.